specification/dartLangSpec.tex

```diff
@@ -18227,14 +18230,20 @@ \section{Libraries and Scripts}
 
 \LMHash{}%
 The \Index{public namespace} of library $L$ is the namespace that maps
-the simple name of each public top-level member declaration $m$ of $L$ to $m$.
+the name of each public top-level member declaration $m$ of $L$ to $m$.
 The \Index{local namespace} of library $L$ is the namespace that maps
 the names introduced by each top-level declaration of $L$
 to the corresponding declaration.
 The \Index{library scope} of library $L$ is the outermost scope in $L$,
 and its namespace is the library namespace of $L$
 (\ref{theImportedNamespace}).
 
+\LMHash{}%
+% A setter can not be paired with a function, class, etc.
+It is a compile-time error if the local namespace of library $L$
+has two declarations with the same basename,
+except when they are a getter and a setter.
```

Comment:
... then a namespace *cannot* hold two members with the same name. This then only catches a setter and a non-getter, but not a getter and a method, or two getters.

Maybe make it a compile-time error for the *library* to have two declarations with the same basename unless one is a getter declaration and the other is a setter declaration.

That catches the conflict "before" creating the namespace, and the namespace can then be treated as injective.

(Or is this handled elsewhere by the general rule that no scope can contain two declarations with the same name, in which case it is fine to only catch the setter + non-getter situation)

Reply:
There is no conflict here: I'm taking into account that the namespace is a partial function, so it does not contain two bindings with the same key.

So "same basename" implies "a pair of names of the form `id` and `id=`.

We can't have a clash between a getter and a method etc., because then the namespace wouldn't be a partial function, and that's already a compile-time error (page 12). So we are already catching the "same name" conflicts before creating the namespace. This extra error is concerned with `id`/`id=` conflicts only.

But this is not easy to see from the text, so we need a clarification. I've added commentary to achieve that.